### DISCUSSION TITLE
What's the recommended way to implement Slack notifications

### DISCUSSION THREAD

**Discussion by airhorns** [Shopify/shipit-engine#1147] [Q&A] [answered]
Hi folks -- I'd love to notify my company's internal Slack when a deploy starts and ends. Digging through issues like https://github.com/Shopify/shipit-engine/issues/1074 and https://github.com/Shopify/shipit-engine/pull/198 shows that Shipit doesn't have this functionality built in which makes a lot of sense.

Is there a current recommended path for getting the notifications out of shipit and into a chatbot? I saw some references to a webhook API but in looking around the codebase and docs didn't see anything about outgoing webhook delivery, just receiving. 

Thanks for any pointers you can give me!

**Comment by casperisfine** [comment, marked as answer]
Hey Harry 👋 

The recommended way is indeed to send webhooks to a chatbot app that will then act as the gateway. 

The outgoing Webhooks are indeed under-documented, and have pretty much no UI... Thet are handled by the [Shipit::Hooks model](https://github.com/Shopify/shipit-engine/blob/cbe2d0c895a39dada5072b06b730f48e526b7969/app/models/shipit/hook.rb), see also [the schema](https://github.com/Shopify/shipit-engine/blob/cbe2d0c895a39dada5072b06b730f48e526b7969/test/dummy/db/schema.rb#L120-L130). Each record is basically a subscription.

So right know users are expected to either create them from a Rails console, or a migration.

Alternatively if you don't have a chat bot, or would rather keep this contained in Shipit, while a bit of a hack, you can decorate the `Shipit::Hook.emit` method in your host app, and use that to implement what you need there. We should probably make this a first class API.

Hope this helps.

**Reply by airhorns** [reply]
Ah ha makes sense! Thanks!

Also -- hi!! 👋 Thanks for makin an awesome open source bit of software!
